### PR TITLE
Python API for capture and streaming with FFmpeg.

### PR DESCRIPTION
**Pull request by SunsetOrange**
This script eases the use of FFmpeg as a stream capture service by abstracting
away complexity and setting a common encoding standard.

### CODE REVIEW

**Comment by Lanzafame** [comment]
I am sorry. I was reading through your pull request and I realised that I have left out several things from the Contributing Guide. I will update the guide tomorrow morning and then comment on your PR what needs to be updated.

File: stream.py [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+script. I suspect that the correct streaming protocol will be rtp/rtsp, with the
+server rebroadcasting the stream over a variety of protocols.
+
+Lots of work to do still. Code is a bit of a mess. Need to make video bitrate
+customizable plus many other things..
+
+ffmpeg streaming guide: https://trac.ffmpeg.org/wiki/StreamingGuide
+"""
+
+__author__ = "Michael Bedford"
```
Comment (Lanzafame) [inline stream.py]: Not to sound rude, but could you remove the author line. I have written out an explanation [here](https://github.com/codenetwork/CodeNetworkTV/wiki/Explanations). 
tl;dr: once someone else makes a contribution its not true anymore.

File: stream.py [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+	import urllib
+elif PY_VERSION[0] == 2:
+	import urllib2
+
+"""
+This script eases the use of FFmpeg as a stream capture service by abstracting
+away complexity and setting a common encoding standard. The stream is sent to
+CODE TV Network to be forwarded onto viewers via FFserver or like server.
+
+The script can be instigated via a command prompt or imported as a module.
```
Comment (Lanzafame) [inline stream.py]: Could you add how-to for installing any required dependencies? Or links to guides for the three main OSs?

File: stream.py [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+
```
Comment (Lanzafame) [inline stream.py]: Could you please add a how-to get started with module? Also your idea/plans for turning this into a independent microservice? Don't feel you need to know how, but express some where what functions/classes need to be exposed to the outside world for it to be useful to the other parts of the system.

**Comment by Lanzafame** [comment]
An minor note, the pull request is too big. Obviously starting a project is a bit different, but for future reference try and keep them below 100-150 lines. That way when someone is reviewing your code, they are less likely to miss an errors due to them getting bored.
Otherwise, awesome work. Cheers for the contribution. :smile:

**Comment by SunsetOrange** [comment]
Sure thing. I haven't got it to my desired state yet, and this week has been a big distraction with guests visiting; but I should be back into the swing of things this weekend. I can almost guarantee there are errors in it and it won’t run (though it was working before I made some changes).
I wanted to get something up to spur some development in other areas.

I'm planning on finishing the code then implementing full docstrings and a rundown of the module. I’ll also upload a readme file or something with some of my deeper thoughts on the module, it’s use and its relationship to the project as a whole.

**Comment by Lanzafame** [comment]
Awesome. Sounds good. Yeah I hoping to get a Github OAuth Client done tonight/tomorrow.
